compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/components/ResolutionParts.kt

```diff
@@ -598,26 +599,27 @@ internal object CheckReceivers : ResolutionPart() {
         resolveKotlinArgument(receiverArgument, receiverParameter, receiverInfo)
     }
 
-    private fun KotlinResolutionCandidate.searchForAdditionalReceivers() {
+    private fun KotlinResolutionCandidate.searchForAdditionalReceivers(): List<SimpleKotlinCallArgument>? {
+        val result = mutableListOf<ReceiverValueWithSmartCastInfo>()
         val candidateReceivers = scopeTower.lexicalScope.parentsWithSelf
             .flatMap { if (it is LexicalScope) scopeTower.getImplicitReceivers(it) else emptyList() }
-        parameterLoop@ for (receiver in candidateDescriptor.additionalReceiverParameters) {
+        for (receiver in candidateDescriptor.additionalReceiverParameters) {
             val expectedReceiverType = receiver.type
             val expectedReceiverTypeClosestBound =
                 if (expectedReceiverType.isTypeParameter()) expectedReceiverType.supertypes().first()
                     .replaceArgumentsWithStarProjections()
                 else expectedReceiverType.replaceArgumentsWithStarProjections()
-
-            for (candidateReceiver in candidateReceivers) {
+            val selectedCandidate = candidateReceivers.firstOrNull { candidateReceiver ->
                 val candidateReceiverType = candidateReceiver.receiverValue.type.replaceArgumentsWithStarProjections()
-                if (NewKotlinTypeChecker.Default.isSubtypeOf(candidateReceiverType, expectedReceiverTypeClosestBound)) {
-                    resolvedCall.additionalReceiversArguments.add(ReceiverExpressionKotlinCallArgument(candidateReceiver))
-                    continue@parameterLoop
-                }
+                NewKotlinTypeChecker.Default.isSubtypeOf(candidateReceiverType, expectedReceiverTypeClosestBound)
             }
-            this.diagnosticsFromResolutionParts.add(NoAdditionalReceiver(receiver))
-            return
+            if (selectedCandidate == null) {
```

Comment:
I'd just return null here (with `?: return null`), but reported an error on the call-site
So, the responsibility of the function would be limited and clear

Reply:
The problem here is that I will lose the information about the missing receiver if I report an error outside of this function.